desc/integrals/basis.py

```diff
@@ -256,14 +257,187 @@ def compute_cheb(self, x):
+        self.Y = truncate_y if (0 < truncate_y < Y) else Y
+        self._c = dctn(f, type=2 - lobatto, axes=(-2, -1))[..., : self.X, : self.Y] / (
+            (X - lobatto) * (Y - lobatto)
+        )
+
+    @staticmethod
+    def nodes(X, Y, L=None, domain_x=(-1, 1), domain_y=(-1, 1), lobatto=False):
+        """Tensor product grid of optimal collocation nodes for this basis.
+
+        Parameters
+        ----------
+        X : int
+            Grid resolution in x direction. Preferably power of 2.
+        Y : int
+            Grid resolution in y direction. Preferably power of 2.
+        L : int or jnp.ndarray
+            Optional, resolution in radial direction of domain [0, 1].
+            May also be an array of coordinates values. If given, then the
+            returned ``coords`` is a 3D tensor-product with shape (L * X * Y, 3).
+        domain_x : tuple[float]
+            Domain for x coordinates. Default is [-1, 1].
+        domain_y : tuple[float]
+            Domain for y coordinates. Default is [-1, 1].
+        lobatto : bool
+            Whether to use the Gauss-Lobatto (Extrema-plus-Endpoint)
+            instead of the interior roots grid for Chebyshev points.
+
+        Returns
+        -------
+        coords : jnp.ndarray
+            Shape (X * Y, 2).
+            Grid of (x, y) points for optimal interpolation.
+
+        """
+        x = cheb_pts(X, domain_x, lobatto)
+        y = cheb_pts(Y, domain_y, lobatto)
+        if L is None:
+            coords = (x, y)
+        else:
+            if isposint(L):
+                L = jnp.flipud(jnp.linspace(1, 0, L, endpoint=False))
+            coords = (jnp.atleast_1d(L), x, y)
+        coords = tuple(map(jnp.ravel, jnp.meshgrid(*coords, indexing="ij")))
+        return jnp.column_stack(coords)
+
+    def evaluate(self, X, Y):
+        """Evaluate Chebyshev series on tensor-product grid.
+
+        Parameters
+        ----------
+        X : int
+            Grid resolution in x direction. Preferably power of 2.
+        Y : int
+            Grid resolution in y direction. Preferably power of 2.
+
+        Returns
+        -------
+        fq : jnp.ndarray
+            Shape (..., X, Y)
+            Chebyshev series evaluated at
+            ``ChebyshevSeries.nodes(X,Y,L,self.domain_x,self.domain_y,self.lobatto)``.
+
+        """
+        warnif(
+            X < self.X,
+            msg="Frequency spectrum of DCT interpolation will be truncated because "
+            "the grid resolution is less than the Chebyshev resolution.\n"
+            f"Got X = {X} < {self.X} = self.X.",
+        )
+        warnif(
+            Y < self.Y,
+            msg="Frequency spectrum of DCT interpolation will be truncated because "
+            "the grid resolution is less than the Chebyshev resolution.\n"
+            f"Got Y = {Y} < {self.Y} = self.Y.",
+        )
+        BUG = jax.__version_info__ <= (0, 7, 2)
+        errorif(
+            self._c.ndim > 2 and BUG,
```

Comment:
this would only be an issue if we're approximating a vector valued qty correct? ie `self._c.shape == (..., X, Y)`?